components/org.wso2.carbon.identity.api.server.notification.sender/org.wso2.carbon.identity.api.server.notification.sender.v2/src/gen/java/org/wso2/carbon/identity/api/server/notification/sender/v2/NotificationSendersApi.java

```diff
@@ -0,0 +1,436 @@
+/*
+ * Copyright (c) 2025, WSO2 LLC. (http://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.api.server.notification.sender.v2;
+
+import org.apache.cxf.jaxrs.ext.multipart.Attachment;
+import org.apache.cxf.jaxrs.ext.multipart.Multipart;
+import java.io.InputStream;
+import java.util.List;
+
+import org.wso2.carbon.identity.api.server.notification.sender.v2.model.EmailSender;
+import org.wso2.carbon.identity.api.server.notification.sender.v2.model.EmailSenderAdd;
+import org.wso2.carbon.identity.api.server.notification.sender.v2.model.EmailSenderUpdateRequest;
+import org.wso2.carbon.identity.api.server.notification.sender.v2.model.Error;
+import org.wso2.carbon.identity.api.server.notification.sender.v2.model.PushSender;
+import org.wso2.carbon.identity.api.server.notification.sender.v2.model.PushSenderAdd;
+import org.wso2.carbon.identity.api.server.notification.sender.v2.model.PushSenderUpdateRequest;
+import org.wso2.carbon.identity.api.server.notification.sender.v2.model.SMSSender;
+import org.wso2.carbon.identity.api.server.notification.sender.v2.model.SMSSenderAdd;
+import org.wso2.carbon.identity.api.server.notification.sender.v2.model.SMSSenderUpdateRequest;
+import org.wso2.carbon.identity.api.server.notification.sender.v2.NotificationSendersApiService;
+import org.wso2.carbon.identity.api.server.notification.sender.v2.factories.NotificationSendersApiServiceFactory;
+
+import javax.validation.Valid;
+import javax.ws.rs.*;
+import javax.ws.rs.core.Response;
+import io.swagger.annotations.*;
+
+import javax.validation.constraints.*;
+
+@Path("/notification-senders")
+@Api(description = "The notification-senders API")
+
+public class NotificationSendersApi  {
+
+    private final NotificationSendersApiService delegate;
```

Comment:
Did we agreed on operating all these methods via a single permission ? `/permission/admin/manage/identity/configmgt/add`

Reply:
The v2 was introduced to change the Email Provider payload and response only. Hence using the permission as same as v1 endpoint.